hugr-py/src/hugr/std/collections/array.py

```diff
@@ -0,0 +1,34 @@
+"""Collection types and operations."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+import hugr.tys as tys
+from hugr.std import _load_extension
+
+EXTENSION = _load_extension("collections.array")
+
+
+@dataclass(eq=False)
+class Array(tys.ExtType):
+    """Fixed `size` array of `ty` elements."""
+
+    ty: tys.Type = field(default_factory=lambda: tys.Unit)
+    size: int = field(default=0)
```

Comment:
i think this stops a type arg being used here?

Reply:
Ah, yes. The parameter should be a `TypeArg` instead, and we cannot store a concrete type.

I'll add some tests